File: apps/api/src/main/java/com/rpg/redsunapi/tale/dto/TaleUpdateRequestDTO.java
package com.rpg.redsunapi.tale.dto;

import com.rpg.redsunapi.tale.enums.ETaleStatus;
import jakarta.validation.constraints.Size;
import org.springframework.lang.Nullable;
import org.springframework.web.multipart.MultipartFile;

import java.util.Optional;

public record TaleUpdateRequestDTO(
  @Nullable
  @Size(max = 120, message = "taleName must be 120 characters or fewer")
  String taleName,
  @Nullable Boolean isPublic,
  @Nullable @Size(max = 4000) String description,
  @Nullable @Size(max = 50) String language,
  Optional<MultipartFile> image,
  @Nullable Boolean removeImage,
  Optional<ETaleStatus> status,
  @Nullable String rules
) {
}

File: apps/web/src/interface/dtos/tale/TaleUpdateRequestDTO.ts
import {ETaleStatus} from "../../enums/ETaleStatus";
import {ERuleSystem} from "../../enums/ERuleSystem";
import {ELanguage} from "../../enums/ELanguage";

export interface TaleUpdateRequestDTO {
  taleName?: string | null;
  isPublic?: boolean | null;
  description?: string | null;
  language?: ELanguage | null;
  image?: File | null;
  removeImage?: boolean | null;
  status?: ETaleStatus | null;
  rules?: ERuleSystem | string | null;
}
